train.rb

```diff
@@ -0,0 +1,85 @@
+require_relative 'company_name.rb'
+require_relative 'instance_counter'
+
+class Train
+  include CompanyName
+  include InstanceCounter
+
+  attr_reader :speed, :number, :wagons  #Предполагая, что от этого класса будут наследоваться не только грузовые и пассажрские поезда, но и другие типы поездов(новые классы), можно удалить поле type
+
+  @@find = Hash.new

```

Comment:
```suggestion
  @@find = {}

```
рекомендованный вариант инициализации пустого хеша